This is to support argument in Chapter 5 about the results

In [2]:
# Import dependencies
import csv

In [3]:
def count_senses(file_path):
    '''
    Function to count and categorize senses from a TSV file.

    Param:
        file_path (str): Path to the input TSV file.

    Returns:
        None
    '''
    with open(file_path, 'r', encoding='utf-8') as tsvfile:
        tsvreader = csv.reader(tsvfile, delimiter='\t')
        header = next(tsvreader)  

        # Initialize counters for KEEP and DELETE
        total_senses_k = 0
        senses_k = 0
        senses_english_k = 0
        none_language_count_k = 0
        senses_2_languages_k = 0
        senses_2_english_k = 0
        others_k = 0
        
        total_senses_d = 0
        senses_d = 0
        senses_english_d = 0
        none_language_count_d = 0
        senses_2_languages_d = 0
        senses_2_english_d = 0
        others_d = 0
        for row in tsvreader:
            synset, lemma, annotation, goodness, confidence, language = row         
            if annotation == 'KEEP':
                total_senses_k += 1

                if confidence == '1':
                    senses_k += 1

                if confidence == '1' and language == 'English':
                    senses_english_k += 1

                if confidence == 'None' and language == 'None':  
                    none_language_count_k += 1

                if confidence == '2':
                    senses_2_languages_k += 1

                if confidence == '2' and 'English' in language:
                    senses_2_english_k += 1

            elif annotation == 'DELETE':
                total_senses_d += 1

                if confidence == '1':
                    senses_d += 1
                if confidence == '1' and language == 'English':
                    senses_english_d += 1

                if confidence == 'None' and language == 'None':  
                    none_language_count_d += 1

                if confidence == '2':
                    senses_2_languages_d += 1

                if confidence == '2' and 'English' in language:
                    senses_2_english_d += 1

        # Calculate 'Others' count outside the loops
#         others_k = total_senses_k - (senses_k + senses_english_k + none_language_count_k + senses_2_languages_k + senses_2_english_k)
#         others_d = total_senses_d - (senses_d + senses_english_d + none_language_count_d + senses_2_languages_d + senses_2_english_d)
    print('KEEP:')
    print(f'Total senses: {total_senses_k}')
    print(f'Suggested by 1 language: {senses_k}')
    print(f'Suggested by 1 language and that language is English: {senses_english_k}')
    print(f'Has None in language and count: {none_language_count_k}')
    print(f'Suggested by 2 languages: {senses_2_languages_k}')
    print(f'Suggested by 2 languages and one of the languages is English: {senses_2_english_k}')
    print(f'Others: {others_k}')

    print('\nDELETE:')
    print(f'Total senses: {total_senses_d}')
    print(f'Suggested by 1 language: {senses_d}')
    print(f'Suggested by 1 language and that language is English: {senses_english_d}')
    print(f'Has None in language and count: {none_language_count_d}')
    print(f'Suggested by 2 languages: {senses_2_languages_d}')
    print(f'Suggested by 2 languages and one of the languages is English: {senses_2_english_d}')
    print(f'Others: {others_d}')

file_path = './data/merged_file_opus.tsv'
count_senses(file_path)


KEEP:
Total senses: 6131
Suggested by 1 language: 4697
Suggested by 1 language and that language is English: 2856
Has None in language and count: 1211
Suggested by 2 languages: 174
Suggested by 2 languages and one of the languages is English: 141
Others: 0

DELETE:
Total senses: 1215
Suggested by 1 language: 1139
Suggested by 1 language and that language is English: 738
Has None in language and count: 67
Suggested by 2 languages: 9
Suggested by 2 languages and one of the languages is English: 8
Others: 0


In [10]:
def count_senses(file_path):
    """
    Funtion to count senses in the csv file
 
    Param:
    file_path (str): Path to the input TSV file.
    
    Returns:
    None
    """
    with open(file_path, 'r', encoding='utf-8') as tsvfile:
        tsvreader = csv.reader(tsvfile, delimiter='\t')
        header = next(tsvreader)  
        senses_k = 0
        senses_d = 0     
        for row in tsvreader:
            synset, lemma, annotation, goodness, confidence, language = row    
            if annotation == 'KEEP':
                if confidence != 'None' and language != 'None':
                    senses_k += 1 
            if annotation == 'DELETE':

                if confidence != 'None' and language != 'None':
                    senses_d += 1
            total = senses_k + senses_d
    print(f'Total sense matched KEEP: {senses_k}')
    print(f'Total sense matched DELETE: {senses_d}')
    print(total)

file_path = './data/merged_file_opus.tsv'
count_senses(file_path)

Total sense matched KEEP: 4920
Total sense matched DELETE: 1148
6068


# Notes
#KEEP

Suggested by 1 language: 3047

Suggested by 1 language and that language is English: 1061

Has None in language and count: 1978

Suggested by 2 languages: 466

Suggested by 2 languages and one of the languages is English: 197
    
#DELETE

Suggested by 1 language: 1215

Suggested by 1 language and that language is English: 373

Has None in language and count: 110

Suggested by 2 languages: 114

Suggested by 2 languages and one of the languages is English: 451